agent-control/src/agent_control/run/k8s.rs

```diff
@@ -66,9 +68,28 @@ impl AgentControlRunner {
             self.k8s_config.namespace.clone(),
         ));
 
+        let opamp_http_builder = if let Some(opamp_config) = &self.opamp {
+            let k8s_secret_provider = K8sSecretProvider::new(k8s_client.clone());
+
+            let secret_path = format!(
+                "{}:{}:{}",
+                self.k8s_config.namespace, self.k8s_config.secret_private_key_name, K8S_KEY_SECRET
+            );
+
+            let secret = k8s_secret_provider.get_secret(&secret_path)?;
```

Comment:
I'd consider moving this to a helper function, also improve naming to explicitly say that we are getting the `private_key`.